### DISCUSSION TITLE
Request for Plotly or Matplotlib Integration , If can.

### DISCUSSION THREAD

**Discussion by peuportier** [open-webui/open-webui#5245] [General]
Dear OpenWebUI Team,

I hope this message finds you well. I want to extend my heartfelt thanks for all the effort you’ve put into this incredible project. I especially appreciate the work you did to integrate Mermaid—it works perfectly!

I was wondering if it would be possible to add support for other graphing libraries like Plotly (for bar graphs and various chart types) or Matplotlib. These would make fantastic additions to your already amazing tools and provide even more flexibility.

If time constraints prevent you from adding this, please let me know how I might assist with the integration. I’d be happy to contribute in any way I can to help make these libraries accessible for use with the LLM models.

Thank you once again for your dedication and hard work.

Best regards,
Aurélien

**Comment by silentoplayz** [comment]
`Matplotlib` is supported already via Pyodide integration.
![image](https://github.com/user-attachments/assets/3866d1b8-662d-4210-8f79-96fbf87fe9a0)

`Plotly` is not a package built in Pyodide according to [Packages built in Pyodide](https://pyodide.org/en/stable/usage/packages-in-pyodide.html).

**Reply by peuportier** [reply]
"Wow, this is wonderful! Thank you so much—I wasn’t expecting this. I truly appreciate the time you've dedicated to this. It would be even more amazing if, one day, the process could be triggered automatically and display the result without needing to press 'Run.' But once again, many thanks for your time. You've truly made my day! Perhaps implementing an action or filter to automatically execute the code and display the result would be a great enhancement."

**Reply by silentoplayz** [reply]
I can't take any credit as I've only contributed an answer and nothing to the code itself. Do note that I've passed along your words of appreciation towards the maintainer as a thoughtful gesture.

> Perhaps implementing an action or filter to automatically execute the code and display the result would be a great enhancement.

This seems like a decent suggestion at first glance and might be worth considering if someone wants to give it a shot. Then again, this `Run Code` function has been created and allows you to run arbitrary Python or Bash code safely in a gVisor sandbox- https://openwebui.com/f/etienneperot/run_code/

If you are looking for an OpenWebUI *tool* to allow the LLM to run its own code, see here instead: https://openwebui.com/t/etienneperot/run_code/
